lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1219,17 +1217,24 @@ struct StoreOpConversion
                              rewriter.create<LLVM::ConstantOp>(
                                  loc, u32Ty, IntegerAttr::get(u32Ty, elemIdx)));
         }
-#ifdef USE_ROCM
-        llWord = bitcast(llWord, valueElemTy);
-#else
         llWord = bitcast(llWord, valArgTy);
-#endif
         std::string constraint =
             (width == 64) ? "l" : ((width == 32) ? "r" : "c");
         asmArgs.emplace_back(llWord, constraint);
       }
 #ifdef USE_ROCM
-      store(asmArgs[vecStart].first, ptrElems[vecStart]);
+      // TODO: need to revisit this when implementing vectorized stores
+      Value llWord = rewriter.create<LLVM::UndefOp>(loc, wordTy);
+      Value elem = valueElems[vecStart];
+      if (elem.getType().isInteger(1))
```

Comment:
`elem.getType().getBitWidth() < 8`?

Reply:
i'm not sure i understand the concern.

this transformation handles 1 bit integer data types (bits) i believe and the next larger data width is 8 bits which does not need special handling.

Reply:
If we are certain `i1` is the only plausible type then I'm fine with existing logic. Just not sure if we'll see `i2~7`.

Reply:
fixed.

you were right - i2-7 are not built in types but users can define custom integer types with those bit widths.